volatility3/framework/symbols/linux/__init__.py

```diff
@@ -265,8 +266,26 @@ def files_descriptors_for_process(
         for fd_num, filp in enumerate(fds):
             if filp != 0:
                 full_path = LinuxUtilities.path_for_file(context, task, filp)
-
-                yield fd_num, filp, full_path
+                dentry = filp.get_dentry()
```

Comment:
Ok, so this is now a non-additive API change, and it would mean that the version number for this needs incrementing to `(3, 0, 0)` on line 64 (not the whole framework @eve-mem but yes, something used by lots of plugins and therefore would require bumping lots of core plugins requirements and potentially break some external plugins).

Another option would be to define an additional function but I'd then only return the `inode_num, imode, ctime, mtime, atime and file_size` since the rest can be recovered from the original call.  It's a little bit of code duplication, but you could actually extract the start of this out to a private method so they'd share the same code?  Doing that would then be an additive change, and the version would only need to go up to `(2, 2, 0)` and all the existing plugins would continue working just fine.

Reply:
Hello @ikelos, 
Thank you for your guidance. 

I'll probably go with option two which indeed implies code reuse but will maximize compatibility with any other existing plugins. 

I'll update the code and propose a new implementation. 

Regards.